src/pt/doramogo/src/eu/kanade/tachiyomi/animeextension/pt/doramogo/Doramogo.kt

```diff
@@ -0,0 +1,224 @@
+
+    override fun popularAnimeSelector() = "div.item-drm"
+
+    override fun popularAnimeFromElement(element: Element) = SAnime.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        title = element.selectFirst("div.title h3")!!.text()
+        thumbnail_url = element.selectFirst("div.cover > img")!!.attr("src")
+    }
+
+    override fun popularAnimeNextPageSelector() = null
+
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/doramas/?filter_orderby=date", headers)
+
+    override fun latestUpdatesSelector() = popularAnimeSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = popularAnimeFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = null
+
+    // =============================== Search ===============================
+    override suspend fun getSearchAnime(page: Int, query: String, filters: AnimeFilterList): AnimesPage {
+        return if (query.startsWith(PREFIX_SEARCH)) { // URL intent handler
+            val id = query.removePrefix(PREFIX_SEARCH)
+            client.newCall(GET("$baseUrl/dorama/$id"))
+                .awaitSuccess()
+                .use(::searchAnimeByIdParse)
+        } else {
+            super.getSearchAnime(page, query, filters)
+        }
+    }
+
+    private fun searchAnimeByIdParse(response: Response): AnimesPage {
+        val details = animeDetailsParse(response.asJsoup())
+        return AnimesPage(listOf(details), false)
+    }
+
+    @Serializable
+    data class SearchResponseDto(
+        val results: List<String>,
+        val page: Int,
+        val total_page: Int = 1,
+    )
+
+    private val searchToken by lazy {
+        client.newCall(GET("$baseUrl/lista-de-animes", headers)).execute()
+            .asJsoup()
+            .selectFirst("div.menu_filter_box")!!
+            .attr("data-secury")
+    }
+
+    override fun getFilterList() = DoramogoFilters.FILTER_LIST
+
+    override fun searchAnimeRequest(page: Int, query: String, filters: AnimeFilterList): Request {
+        val params = DoramogoFilters.getSearchParameters(filters)
+
+        val url = "$baseUrl/search/$query".toHttpUrl().newBuilder()
+            .addIfNotBlank("filter_audio", params.audio)
+            .addIfNotBlank("filter_genre", params.genre)
+            .build()
+            .toString()
+
+        return GET(url, headers = headers)
+    }
+
+    override fun searchAnimeParse(response: Response): AnimesPage {
+        return runCatching {
+            val data = response.parseAs<SearchResponseDto>()
+            val animes = data.results.map(Jsoup::parse)
+                .mapNotNull { it.selectFirst(searchAnimeSelector()) }
+                .map(::searchAnimeFromElement)
+            val hasNext = data.total_page > data.page
+            AnimesPage(animes, hasNext)
+        }.getOrElse { AnimesPage(emptyList(), false) }
```

Comment:
This is going to silently catch all errors, I'd say catching a more specific exception and then throwing an exception with a message to inform the user in some way